src/daft-io/src/object_store_glob.rs

```diff
@@ -404,6 +404,29 @@ pub async fn glob(
     };
     let glob = glob.as_str();
 
+    // We need to do some validation on the glob pattern before compiling it, since the globset crate is very permissive
+    // and will happily compile patterns that don't make sense without throwing an error.
+    fn verify_glob(glob: &str) -> super::Result<()> {
+        // Catch for cases like `s3://bucket/path/**.txt`
+        // NOTE: "\**" is a valid pattern that matches a literal `*`, followed by anything, so we need to only capture cases where `**` is not preceded by a backslash
+        let re = regex::Regex::new(r"(?:[^\\]|^)\*\*").unwrap();
+
+        for segment in glob.split(GLOB_DELIMITER) {
+            if re.is_match(segment) && segment != "**" {
+                return Err(super::Error::InvalidArgument {
+                    msg: format!(
+                        "Invalid usage of '**' in glob pattern. The '**' wildcard must occupy an entire path segment and be surrounded by '{}' characters. Found invalid usage in '{}'.",
```

Comment:
Could we be more helpful with the error message as well? Would love to add a suggestion here for the user to do this glob path instead: `{re_group_1}/**/*{re_group_2}/{re_group_3}`.

Reply:
Good point. I've slightly rewritten the regex to process the path fully instead of segmenting it delimited portions. 
This should allow us to suggest a corrected glob path for the user as well.

Have rewritten it to give suggestions in this manner: is this the behaviour you expect?
- **Original:** `invalid/blahblah**.txt` → **Corrected:** `invalid/blahblah/**/*.txt`
- **Original:** `invalid/\***.txt` → **Corrected:** `invalid/\\*/**/*.txt`
- **Original:** `invalid/\**blahblah**.txt` → **Corrected:** `invalid/\\**blahblah/**/*.txt`